### PR TITLE
Add Pebble test container for Let's Encrypt with CA cert

### PR DESCRIPTION
**Pull request by humphd**
This adds the Let's Encrypt test server, [pebble](https://github.com/letsencrypt/pebble), do our `docker-compose.yml`.  I've currently set it up to always approve challenges (i.e., run without a challenge server, but we can add that later if we want (see example in https://github.com/Seneca-CDOT/starchart/pull/3/files#diff-e45e45baeda1c1e73482975a664062aa56f20c03dd9d64a827aba57775bed0d3R70).

I've also included the [Pebble CA cert](https://github.com/letsencrypt/pebble/blob/main/test/certs/localhost/cert.pem) in `test/resources/ca.cert.pem`.  We can add this to node when we run tests doing:

`cross-env NODE_EXTRA_CA_CERTS=./test/resources/ca.cert.pem node ...`

I've included a note in the `docker-compose.yml` on how to do this.

### CODE REVIEW

**Comment by dadolhay** [comment]
How will the system know that `test/resources/ca.cert.pem` should be available under `secrets.LETS_ENCRYPT_ACCOUNT_PRIVATE_KEY_PEM`? can we map it somehow or do we rename something?

**Comment by humphd** [comment]
You can start node with extra certs by doing `NODE_EXTRA_CA_CERTS=./test/resources/ca.cert.pem node test` or something (i.e., it will read the `PEM` file at that path and add to its list of accepted certs).

**Comment by dadolhay** [comment]
So this is actually the root certificate that the mock let's encrypt will use to sign the requested ssl certs?

**Comment by dadolhay** [comment]
Okay, sorry, I haven't seen that part of the description.
The docker has this file specified: `/test/config/pebble-config.json`
Will we add that later? or is that in the container? How will Pebble know the use that CA cert?

**Comment by humphd** [comment]
`/test/config/pebble-config.json` is part of the image, see https://github.com/letsencrypt/pebble/blob/main/test/config/pebble-config.json

**Comment by dadolhay** [comment]
Okay. so the cert has the following info:
```
Common Name: minica root ca 24e2db
Valid From: December 6, 2017
Valid To: December 6, 2117
Serial Number: 2657928050610294462 (0x24e2db7acf2c4ebe)
```
so it should not expire for a long time.

What's still not 100% clear is, that the docs also state:
```
Note that the CA's root and intermediate certificates are regenerated on every launch. 
```
Regardless, I think this one is good to go. Thank you for the explanation!

**Review by dadolhay — APPROVED** [review]

**Comment by humphd** [comment]
Yeah, the signing cert isn't useful beyond testing in the current session.  But to have node allow `fetch()` etc. using `https://` with the container, we have to trust it during the test run.  Maybe this is already clear, but in case not for others reviewing, that's the point of this extra PEM.

**Comment by dadolhay** [comment]
Okay, I get it now.

So this minica certificate is only to communicate between nodejs and the pebble server. It has nothing to do with signing the actual certificates that the server issues. That CA and Intermediate CA is generated at startup as the docs state.

**Review by Genne23v — APPROVED** [review]
Looks good to me

File: docker-compose.yml [hunk 1]
```diff
@@ -16,3 +16,21 @@ services:
       MYSQL_USER: starchart
       MYSQL_PASSWORD: starchart_password
       MYSQL_ROOT_PASSWORD: root_password
+
+  # Let's Encrypt test server - https://github.com/letsencrypt/pebble
+  # Extra SSL Cert for testing available at: `test/resources/ca.cert.pem`
+  # To use in code/tests, add to node's certificate chain via env var:
+  # `cross-env NODE_EXTRA_CA_CERTS=./test/resources/ca.cert.pem node ...`
+  lets_encrypt:
+    image: letsencrypt/pebble:latest
```
[the remark this answers is not in the document]
Comment (humphd) [reply docker-compose.yml:25]: Looks like it's filed upstream https://github.com/letsencrypt/pebble/issues/394.  I think we're probably OK for now, but I'll watch that issue.  Great catch.

**Comment by humphd** [comment]
Thanks for the reviews, merged.

**Pull request closed without merging**